src/Arcus.Messaging.Pumps.Abstractions/Extensions/IServiceCollectionExtensions.cs

```diff
@@ -28,6 +29,7 @@ public static class IServiceCollectionExtensions
             Guard.NotNull(implementationFactory, nameof(implementationFactory), "Requires a factory implementation function to create the custom message pump instance");
 
             services.TryAddSingleton<IMessagePumpLifetime, DefaultMessagePumpLifetime>();
+            services.TryAddSingleton<IMessagePumpCircuitBreaker, DefaultMessagePumpCircuitBreaker>();
```

Comment:
Yes, you have a point there. There is some shared functionality. We could maybe see if the implementation could be in the same class, but that we expose it as different interfaces?

Reply:
Maybe too early to decide on this, but my feeling is that we'll depricate the `IMessagePumpLifetime` interface in the end.

Reply:
Yes, that could also be an option; especially since the circuit breaker functionality is more interesting. I also looked at the background jobs repo, and we don't seem to use it there either, so that's good. Not sure if this is used much, to be honest.